app/assets/javascripts/discourse/app/components/emoji-picker.js

```diff
@@ -147,9 +137,10 @@ export default Component.extend({
       // of blocking the rendering of the picker
       discourseLater(() => {
         schedule("afterRender", () => {
+          document.addEventListener("click", this.handleOutsideClick);
```

Comment:
This change is the core of this fix. It defers the registering of the "click" event handler after the `EmojiPicker` has the time to render. Otherwise, the click on the `more...` option in the emoji autocomplete was considered as a click outside the emoji picker and would immediately close the emoji picker...